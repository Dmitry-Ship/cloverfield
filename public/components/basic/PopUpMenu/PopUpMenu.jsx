import React, { PropTypes } from 'react';

import { item, item__link, popupMenu, list } from './PopUpMenu.styl';

import { Link } from 'react-router';

import List from '../List';

const PopUpMenu = ({ items, className, children }) => {
  const customRender = ({ func, href, text }, i) => {
    return (
      <li className={item} key={i} onMouseDown={func}>
        {href
          ? <Link className={item__link} to={href}>{text}</Link>
          : text}
      </li>
    )
  }

  return (
    <div className={`${popupMenu} ${className ? className : ""}`}  >
      {children}
      {items && <List className={list} items={items} itemRenderer={customRender}/>}
    </div>
  )
}

export default PopUpMenu;

PopUpMenu.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
  children: PropTypes.element,
  className: PropTypes.string

}
