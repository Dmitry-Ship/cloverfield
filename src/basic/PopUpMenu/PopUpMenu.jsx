import React, { PropTypes } from 'react';

import { Link } from 'react-router';
import List from '../List';

const PopUpMenu = ({ items, className, children }) => {
  const customRender = (li, i) => {
    return (
      <li className="popup-menu__item" key={i} onMouseDown={li.func}>
        {li.href
          ? <Link className="popup-menu__item__link" to={li.href}>{li.text}</Link>
          : li.text}
      </li>
    )
  }

  return (
    <div className={`popup-menu ${className ? className : ""}`}  >
      {children}
      {items && <List className="popup-menu__list" items={items} itemRenderer={customRender}/>}
    </div>
  )
}

export default PopUpMenu;

PopUpMenu.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
  children: PropTypes.element,
  className: PropTypes.string

}
