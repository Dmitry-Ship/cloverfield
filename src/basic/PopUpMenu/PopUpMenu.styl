.popup-menu
  padding 10px 0
  fancy-shadow()
  color black
  z-index: 2
  // +below(600px)
  //   &::after, ::before
  //     display: none;
  //   fixed top 50% left 50%
  //   transform translate(-50%, -50%) scale(2)

.popup-menu__list
  padding: 0
  margin: 0

.popup-menu__item
  display block
  padding 12px 15px
  white-space nowrap
  border-left 3px solid transparent
  border-right 3px solid transparent
  cursor pointer
  &:hover
    color $color__accent
    border-left: 3px solid $color__accent
    border-right: 3px solid $color__accent
    .popup-menu__item__link
      color $color__accent

.popup-menu__item__link
  color black
