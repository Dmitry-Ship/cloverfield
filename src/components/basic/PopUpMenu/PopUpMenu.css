

.popupMenu {
	z-index: 2;
	padding: 10px 0;
	color: #000;
}

.list {
	margin: 0;
	padding: 0;
}

.item {
	display: block;
	padding: 12px 15px;
	border-right: 3px solid transparent;
	border-left: 3px solid transparent;
	white-space: nowrap;
	cursor: pointer;
}

.item:hover {
	border-right: 3px solid var(--color__accent);
	border-left: 3px solid var(--color__accent);
	color: var(--color__accent);
}

.item:hover .item__link {
	color: var(--color__accent);
}

.item__link {
	color: #000;
}
