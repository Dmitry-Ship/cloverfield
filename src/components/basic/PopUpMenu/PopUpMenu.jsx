import React, { PropTypes } from 'react';

import styles from './PopUpMenu.css';

import { Link } from 'react-router';
import List from '../List';

const PopUpMenu = ({ items, className, children }) => {
  const customRender = (li, i) => {
    return (
      <li className={styles.item} key={i} onMouseDown={li.func}>
        {li.href
          ? <Link className={styles.item__link} to={li.href}>{li.text}</Link>
          : li.text}
      </li>
    )
  }

  return (
    <div className={`${styles.popupMenu} ${className ? className : ""}`}  >
      {children}
      {items && <List className={styles.list} items={items} itemRenderer={customRender}/>}
    </div>
  )
}

export default PopUpMenu;

PopUpMenu.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
  children: PropTypes.element,
  className: PropTypes.string

}
