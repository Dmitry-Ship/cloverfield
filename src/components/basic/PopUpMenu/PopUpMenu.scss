@value colors: "../../../styles/colors.scss";
@value accent, secondary from colors;

$mainColor: white;
$pointerSize: 10px;

.popupMenu {
  display: none;
  composes: level-3 from 'styles/shadows.scss';
  color: black;
  z-index: 9;
  position: absolute;
  text-align: center;
  background: $mainColor;
  &::after, ::before {
    content: "";
    border: $pointerSize solid;
    position: absolute;
  }
}

.bottom {
  composes: popupMenu;
  transform: translateX(-50%);
  top: calc(100% + #{$pointerSize});
  left: 50%;
  &::after, ::before {
    bottom: 100%;
    left: 50%;
    transform: translateX(-50%);
    border-color: transparent transparent $mainColor transparent;
  }
}

.topRight {
  composes: popupMenu;
  transform: translateX(-14%);
  bottom: calc(100% + #{$pointerSize});
  left: 50%;
  &::after, ::before {
    top: 100%;
    left: 14%;
    transform: translateX(-50%);
    border-color: $mainColor transparent transparent transparent;
  }
}

.top {
  composes: popupMenu;
  transform: translateX(-50%);
  bottom: calc(100% + #{$pointerSize});
  left: 50%;
  &::after, ::before {
    top: 100%;
    left: 50%;
    transform: translateX(-50%);
    border-color: $mainColor transparent transparent transparent;
  }
}

.right {
  composes: popupMenu;
  transform: translateY(-50%);
  left: calc(100% + #{$pointerSize});
  top: 50%;
  &::after, ::before {
    top: 50%;
    right: 100%;
    transform: translateY(-50%);
    border-color: transparent $mainColor transparent transparent;
  }
}

.left {
  composes: popupMenu;
  transform: translateY(-50%);
  right: calc(100% + #{$pointerSize});
  top: 50%;
  &::after, ::before {
    top: 50%;
    left: 100%;
    transform: translateY(-50%);
    border-color: transparent transparent transparent $mainColor;
  }
}
