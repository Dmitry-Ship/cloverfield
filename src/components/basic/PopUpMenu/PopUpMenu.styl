@import '../../../styles/index.styl'

.popupMenu
  padding 10px 0
  color black
  z-index: 2
  fancy-shadow()

.list
  padding: 0
  margin: 0

.item
  display block
  padding 12px 15px
  white-space nowrap
  border-left 3px solid transparent
  border-right 3px solid transparent
  cursor pointer
  &:hover
    color color__accent
    border-left: 3px solid color__accent
    border-right: 3px solid color__accent
    .item__link
      color color__accent

.item__link
  color black
