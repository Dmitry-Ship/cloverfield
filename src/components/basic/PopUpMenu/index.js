export { default } from './PopUpMenu.jsx';
