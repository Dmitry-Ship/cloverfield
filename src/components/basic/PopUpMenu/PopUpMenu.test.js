import React from 'react';
import renderer from 'react-test-renderer';
import PopUpMenu from './PopUpMenu';

describe('<PopUpMenu />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
        <PopUpMenu >Hello World</PopUpMenu>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with position passed in', () => {
    const tree = renderer.create(
        <PopUpMenu position={'top'} >Hello World</PopUpMenu>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});