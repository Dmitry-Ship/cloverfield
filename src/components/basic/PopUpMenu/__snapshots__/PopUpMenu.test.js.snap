// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopUpMenu /> should render correctly 1`] = `
<div
  className="top "
>
  Hello World
</div>
`;

exports[`<PopUpMenu /> should render correctly with position passed in 1`] = `
<div
  className="top "
>
  Hello World
</div>
`;
